lib/fs/fakefs_test.go

```diff
@@ -172,3 +178,722 @@ func TestFakeFSRead(t *testing.T) {
 		t.Error("data mismatch")
 	}
 }
+
+type testFS struct {
+	name string
+	fs   Filesystem
+}
+
+type test struct {
+	name string
+	impl func(t *testing.T, fs Filesystem)
+}
+
+func TestFakeFSCaseSensitive(t *testing.T) {
+	var tests = []test{
+		{"RemoveAll", testFakeFSRemoveAll},
+		{"Remove", testFakeFSRemove},
+		{"SameFile", testFakeFSSameFile},
+	}
+	var filesystems = []testFS{
+		{"fakefs", newFakeFilesystem("/foo")},
+	}
+
+	if runtime.GOOS == "linux" {
+		if err := os.Mkdir("test-tmp", 0755); err != nil {
```

Comment:
See the code I posted, we should use that to run tests in a temp dir rather than some specific dir in our local checkout (which then if fails makes git checkout dirty)